api/v1beta1/azuremachine_types.go

```diff
@@ -142,6 +142,12 @@ type AzureMachineSpec struct {
 	// The primary interface will be the first networkInterface specified (index 0) in the list.
 	// +optional
 	NetworkInterfaces []NetworkInterface `json:"networkInterfaces,omitempty"`
+
+	// CapacityReservationGroupID specifies the capacity reservation group resource id that should be
+	// used for allocating the virtual machine.
+	// +kubebuilder:validation:nullable
```

Comment:
Why is this validation needed? For a string pointer, I think the `omitempty` should handle any null inputs.

Reply:
Initially, I had used `armcompute.SubResource` as the type for `CapacityReservationGroupID`. Later changed it to string pointer. Forgot to remove `// +kubebuilder:validation:nullable`. Thanks for pointing it out.